sample-action-client/src/extension.ts

```diff
@@ -20,6 +20,18 @@ export async function activate(context: vscode.ExtensionContext) {
         vscode.window.showInformationMessage(`Action ${action.name} was scheduled to run`,);
         basAPI.actions.performAction(action, {schedule: true});
     }));
+
+	context.subscriptions.push(vscode.commands.registerCommand("get.parameter", async () => {
+        const parameterName = await vscode.window.showInputBox({ prompt: "Enter Parameter Name", ignoreFocusOut: true });
+        if (!parameterName) { 
+            // undefined in escape and empty on enter
+            return;
+        }        
+        
+        const parameterValue = await basAPI.getParameter(parameterName as string);
```

Comment:
I think the explicit casting here is not "at the correct spot".

Option1:
- add the casting where you defined `parameterName` (what is the return type of `showInputBox` ?)

Option2:
- extract the `if (!parameterName) {` block to a helper function with a **type guard** 
- https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

we can discuss which option is better offline

Reply:
I'll remove the casting. I don't really need it now. Don't remember why I added it.